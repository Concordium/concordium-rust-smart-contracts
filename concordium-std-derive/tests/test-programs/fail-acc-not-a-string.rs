//! Ensure that the macros generate compilable code.

use concordium_contracts_common::*;
use concordium_std_derive::*;

const ACC: AccountAddress = account_address!(0);

fn f() {
    println!("{:?}", ACC.to_string());
}

fn main() {}
