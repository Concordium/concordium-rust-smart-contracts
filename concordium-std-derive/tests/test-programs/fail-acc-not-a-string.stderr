error: Expected a string literal of a hex-encoded account address
 --> tests/test-programs/fail-acc-not-a-string.rs:6:29
  |
6 | const ACC: AccountAddress = account_address!(0);
  |                             ^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `account_address` (in Nightly builds, run with -Z macro-backtrace for more info)
